### DISCUSSION TITLE
Picking Loihi hardware on cloud connection

### DISCUSSION THREAD

**Discussion by ahmetozleraktas** [lava-nc/lava#441] [Q&A]
Hello everyone, I was looking for a documentation to use one of loihi cards for our Izhikevich neuron. We connected to the system but did not get any results. Code works on host machine (as a Jupyter Notebook) but how can we be sure about hardware computed it? Best regards.

**Comment by awintel** [comment]
Are you saying that you have developed an Izhikevich neural Process with a PyProcModel that executes on the host CPU and now you would like to execute this model on a Loihi system (loihi card)?
If that is what you are asking, then the first thing you have to do is write an NcProcModel including the corresponding micro coded neuron model to execute on Loihi unless you have done this already....

**Reply by ahmetozleraktas** [reply]
Yes, I meant what you said. But I didn't create NcProcModel. Is there a guide or resource for this?

**Comment by awintel** [comment]
First of all, you need to become a member of the Intel Neuromorphic Research Community to gain access to the _Loihi extension for Lava_ if you aren't already. You can learn more about this here:
https://intel-ncl.atlassian.net/wiki/spaces/INRC/overview

We do not yet have a dedicated tutorial for writing microcode neuron models (will come soon). However, there are several examples you can examine. The Sigma Delta Neuron Model that's used as part of the PilotNet tutorial is one of them. You can find it in 
/src/lava/proc/sdn
of the _Loihi extension for Loihi_.

**Reply by ahmetozleraktas** [reply]
https://lava-nc.org/lava-lib-dl/netx/notebooks/pilotnet_sdnn/run.html?highlight=pilotnet

Apart from the example you mentioned, in the link I gave above, I found this `if loihi2_is_available:
    run_config = CustomHwRunConfig()
else:
    run_config = CustomSimRunConfig()`. I didn't see the model you mentioned used but it says it works on hardware in this tutorial. I can view the cards found on the host machine with the sinfo command, but when I check it with the following code below , I cannot verify that it is available.

```
from lava.utils.system import Loihi2
Loihi2.preferred_partition = 'kp_stack'
loihi2_is_available = Loihi2.is_loihi2_available

if loihi2_is_available:
    print(f'Running on {Loihi2.partition}')
    compression = io.encoder.Compression.DELTA_SPARSE_8
else:
    print("Loihi2 compiler is not available in this system. "
          "This tutorial will execute on CPU backend.")
    compression = io.encoder.Compression.DENSE
```
